glean-core/src/metrics/timing_distribution.rs

```diff
@@ -576,6 +577,137 @@ impl TimingDistributionMetric {
             test_get_num_recorded_errors(glean, self.meta(), error).unwrap_or(0)
         })
     }
+
+    /// Start a new histogram buffer associated with this timing distribution metric.
+    ///
+    /// A histogram buffer accumulates in-memory.
+    /// Data is recorded into the metric on drop.
+    pub fn start_buffer(&self) -> LocalTimingDistribution<'_> {
+        LocalTimingDistribution::new(self)
+    }
+
+    /// Start a new histogram buffer.
+    ///
+    /// A histogram buffer accumulates in-memory.
+    /// Data needs to be recorded into the metric using
+    /// [`TimingDistributionMetric::ffi_commit_buffer`].
+    #[doc(hidden)]
+    pub fn ffi_start_buffer(&self) -> Arc<FfiLocalTimingDistribution> {
+        Arc::new(FfiLocalTimingDistribution::new())
+    }
+
+    /// Record data from the histogram buffer into this metric.
+    pub fn ffi_commit_buffer(&self, buffer: Arc<FfiLocalTimingDistribution>) {
```

Comment:
Should we clear the buffer to stop folks accidentally committing the same buffer twice? Or is that prevented somewhere/somehow else? For non-FFI we rely on the "called at most once" behaviour of drop, but I don't see us accounting for the lack of that guarantee (or testing that case) here.